### PR TITLE
docs: describe devcontainers as deployment model

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/12429

This PR describes a deployment model with the workspace running on _envbuilder_ stack.

### CODE REVIEW

File: docs/about/architecture.md [hunk 1]
```diff
@@ -268,3 +268,53 @@ Coder on Kubernetes.
...
+
+##### Workload resources
+
+**Workspace**
+
+- Docker and Kubernetes based templates are supported.
+- The `docker_container` resource uses `ghcr.io/coder/envbuilder` as base image.
+
+The _envbuilder_ checks out the base Docker image from the container registry,
+and applies extra feature layers, and starts the container with target
```
Comment (mafredri) [inline docs/about/architecture.md:311]:
```suggestion
and applies extra feature layers, and starts the container with the target
```

Nit: I'm not sure what this is trying to convey. What's the "target environment"? As a reader I might also wonder what these extra feature layers are.

Perhaps target environment means environment as specified in `devcontainer.json`? If so, I think we could improve the wording.

Note: The dev container specification doesn't mention layers in any significant proportion. There's a tiny note down on this page, for instance: https://containers.dev/implementors/features/. Hence the concept may not be clear. Perhaps "layer" is more of an implementation detail, not something users need to care about/know? (Fine to keep in the graph/image though.)

File: docs/about/architecture.md [hunk 2]
```diff
@@ -268,3 +268,53 @@ Coder on Kubernetes.
...
+environment.
+
+##### Workload supporting resources
+
+**Container Registry (optional)**
+
+- Workspace nodes need access to the Container Registry to check out images. To
+  shorten the provisioning time, it is recommended to deploy regional replicas
+  of the registry.
```
Comment (mafredri) [inline docs/about/architecture.md:320]:
```suggestion
  shorten the provisioning time, it is recommended to deploy registry
  mirrors in the same region as the workspace nodes.
```

s/mirrors/replicas/ dealers choice. To me regional sounded a bit vague, but if you feel the original is better that's fine too.

**Review by mafredri — COMMENTED** [review]
ETOOMANYNITS, but hope you can wade through. Great job 👍🏻

File: docs/about/architecture.md [hunk 3]
```diff
@@ -268,3 +268,53 @@ Coder on Kubernetes.
...
   [Google Cloud Identity Platform](https://cloud.google.com/architecture/identity/single-sign-on)
+
+### Devcontainers
+
+Note: _Devcontainers_ are at early stage and considered experimental at the
+moment.
+
+This architecture enhances a Coder workspace with
+[Devcontainers](https://containers.dev/) setup built using
+[envbuilder](https://github.com/coder/envbuilder) project. Workspace users have
```
[the remark this answers is not in the document]
Comment (mtojek) [reply docs/about/architecture.md]: I will stick close to `Dev container` :) Thanks for pointing it out!

**Review by mafredri — APPROVED** [review]

File: docs/about/architecture.md (hunk 1 above)
Comment (mtojek) [reply docs/about/architecture.md]: Good call, rephrased! I used the word layers as this is mostly what envbuilder does. It builds next layers = features on top of the base image.

File: docs/about/architecture.md [hunk 4]
```diff
@@ -268,3 +268,54 @@ Coder on Kubernetes.
...
+##### Workload resources
+
+**Workspace**
+
+- Docker and Kubernetes based templates are supported.
+- The `docker_container` resource uses `ghcr.io/coder/envbuilder` as the base
+  image.
+
+_Envbuilder_ checks out the base Docker image from the container registry and
+installs selected features as specified in the `devcontainer.json` on top.
```
Comment (mafredri) [inline docs/about/architecture.md:312]: Btw, this update is beautiful, super clear. ❤️

**Pull request merged**